src/coder/records/AAAARecord.ts

```diff
@@ -15,16 +15,22 @@ export class AAAARecord extends ResourceRecord {
   constructor(header: RecordRepresentation, ipAddress: string);
   constructor(name: string | RecordRepresentation, ipAddress: string, flushFlag?: boolean, ttl?: number) {
     if (typeof name === "string") {
-      super(name, RType.AAAA, ttl || AAAARecord.RR_DEFAULT_TTL_SHORT, flushFlag);
+      super(name, RType.AAAA, ttl || AAAARecord.DEFAULT_TTL, flushFlag);
     } else {
       assert(name.type === RType.AAAA);
       super(name);
     }
 
-    assert(net.isIPv6(ipAddress), "IP address is not in v6 format!");
+    // Enhanced validation to check for IPv6 and IPv4-mapped IPv6 addresses
+    assert(net.isIPv6(ipAddress) || this.isIPv4MappedIPv6(ipAddress), "IP address is not in v6 or IPv4-mapped v6 format!");
     this.ipAddress = ipAddress;
   }
-
+  // Utility method to check for IPv4-mapped IPv6 addresses
+  private isIPv4MappedIPv6(ipAddress: string): boolean {
+    const ipv4MappedIPv6Regex = /^::ffff:(0{1,4}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])$/i;
```

Comment:
My suggestion is to rewrite this one to make it more concise/elegant:

```
private isIPv4MappedIPv6(ipAddress: string): boolean {

return /^::ffff:(\d{1,3}\.){3}\d{1,3}$/i.test(address);
}
```

This matches the prefix, then the first three octets of an IP which end in periods, followed by the last octet which doesn't have a period. I've also made it case-insensitive here for good measure and robustness.

Reply:
I commented out what I had and added your changes. I will test this with an hap-nodes alpha and Homebridge alpha release.